src/commonMain/kotlin/exchange.dydx.abacus/state/internalstate/InternalState.kt

```diff
@@ -137,22 +137,42 @@ internal data class InternalSubaccountState(
     var assetPositions: Map<String, InternalAssetPositionState>? = null,
     var subaccountNumber: Int,
     var address: String? = null,
-    var equity: String? = null,
-    var freeCollateral: String? = null,
+    var equity: Double? = null,
+    var freeCollateral: Double? = null,
     var marginEnabled: Boolean? = null,
     var updatedAtHeight: String? = null,
     var latestProcessedBlockHeight: String? = null,
 
-    // Calculate:
-    var quoteBalance: MutableMap<CalculationPeriod, Double?> = mutableMapOf(),
+    var pendingPositions: List<InternalPerpetualPendingPosition>? = null,
 
+    // for parent subaccount only.  This contains the consolidated open positions of all child subaccounts
+    var childSubaccountOpenPositions: Map<String, InternalPerpetualPosition>? = null,
+
+    // Calculated:
+    val calculated: MutableMap<CalculationPeriod, InternalSubaccountCalculated> = mutableMapOf(),
 ) {
+    val isParentSubaccount: Boolean
+        get() = subaccountNumber < NUM_PARENT_SUBACCOUNTS
+
     val openPositions: Map<String, InternalPerpetualPosition>?
-        get() {
-            return positions?.filterValues { it.status == IndexerPerpetualPositionStatus.OPEN }
-        }
+        get() = positions?.filterValues { it.status == IndexerPerpetualPositionStatus.OPEN }
+
+    val groupedOpenPositions: Map<String, InternalPerpetualPosition>?
+        get() = if (isParentSubaccount) childSubaccountOpenPositions else openPositions
 }
 
+internal data class InternalSubaccountCalculated(
+    var quoteBalance: Double? = null,
```

Comment:
QQ: quoteBalance isn't calculated? At least not until there is multi-collateral?

Reply:
Actually, is it in calculated because we get back an AssetPosition and this is the derived state?

Reply:
It's calculated from `SubaccountTransformer`.  See the function `applyDeltaToSubaccount`